### PR TITLE
feat: Allow id or event_id in project events endpoint

### PR DESCRIPTION
**Pull request by alex-hofsteede**
Reprise of a previous PR that failed because there is no index
on `event_id` alone. Now I've added this functionality to the project
event details endpoint where we know we always have the project id
available, and I've made the fronted code use this endpoint instead
of the bare `/api/0/events/` endpoint.

### CODE REVIEW

File: src/sentry/api/endpoints/project_event_details.py [hunk 1]
```diff
@@ -72,7 +104,17 @@ def get(self, request, project, event_id):
...
+            user_report = UserReport.objects.get(
+                event_id=event.event_id,
+                project=event.project,
+            )
+        except UserReport.DoesNotExist:
+            user_report = None
+
         data = serialize(event, request.user)
+        data['userReport'] = serialize(user_report, request.user)
+        data['release'] = self._get_release_info(request, event)
```
Comment (alex-hofsteede) [inline src/sentry/api/endpoints/project_event_details.py]: This `userReport` and `release` stuff is just to make the payload returned by this endpoint the same as the `/events/` one

File: src/sentry/api/endpoints/project_event_details.py [hunk 2]
```diff
@@ -23,6 +24,20 @@ def retrieve_event_for_project_scenario(runner):
 class ProjectEventDetailsEndpoint(ProjectEndpoint):
     doc_section = DocSection.EVENTS
 
+    def _get_release_info(self, request, event):
```
Comment (alex-hofsteede) [inline src/sentry/api/endpoints/project_event_details.py]: copypasta from https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/event_details.py#L18-L30

File: src/sentry/api/endpoints/project_event_details.py (hunk 1 above)
Comment (macqueen) [reply src/sentry/api/endpoints/project_event_details.py]: would probably be nice to move the shared stuff between this and the other endpoint into a serializer or something (maybe `DetailedEventSerializer`?) but doesn't need to be part of this pr

**Review by macqueen — APPROVED** [review]

File: src/sentry/api/endpoints/project_event_details.py (hunk 1 above)
Comment (alex-hofsteede) [reply src/sentry/api/endpoints/project_event_details.py]: Good idea. Have done it here. https://github.com/getsentry/sentry/pull/10921

**Pull request merged**